transformer/events_creator.go

```diff
@@ -36,18 +36,27 @@ type Transformer struct {
 	eval          evaluator.Evaluator
 	eventMetadata common.MapStr
 	events        []beat.Event
+	cm 			  CommonData
 }
 
-func NewTransformer(ctx context.Context, eval evaluator.Evaluator, index string) Transformer {
+func NewTransformer(ctx context.Context, eval evaluator.Evaluator, index string) (Transformer, error) {
 	eventMetadata := common.MapStr{libevents.FieldMetaIndex: index}
 	events := make([]beat.Event, 0)
+	cm := NewCommonData()
+
+	err := cm.fetchCommonData()
```

Comment:
Should we pass a context as well? instead of generating a new one in `common_data.go`.